lib/aytests/iso_repo.rb

```diff
@@ -35,11 +37,20 @@ def initialize(dir)
     # @param [String] uri URI
     # @return [Pathname]  Path to the local ISO.
     def get(uri)
-      iso_path = @dir.join(File.basename(uri))
-      if iso_path.exist?
-        iso_path
-      elsif download_to(uri, iso_path)
-        iso_path
+      iso_path = File.join(@dir, URI(uri).host, URI(uri).path)
+      iso_dir = File.dirname(iso_path)
+      if uri.include?("*") || uri.include?("?")
+        # We have wildcards in the uri. So we have to remove
+        # old ISOs before because the name could have been changed
+        # meanwhile.
+        Dir.entries(iso_dir).each do |filename|
+          path = File.join(iso_dir, filename)
+          File.delete(path) unless File.directory?(path)
+        end
```

Comment:
Oh, because `iso_path` and `iso_dir` are strings instead of pathnames:

```ruby
iso_path = @dir.join(URI(uri).host, URI(uri).path)
iso_dir = iso_path.dirname
```